Api/PaymentTokenRepositoryInterface.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Paytrail\PaymentService\Api;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Vault\Api\Data\PaymentTokenInterface;
+use Magento\Vault\Api\PaymentTokenRepositoryInterface as BasePaymentTokenRepositoryInterface;
+
+interface PaymentTokenRepositoryInterface extends BasePaymentTokenRepositoryInterface
```

Comment:
Please consider renaming the interface and revising your code:
One of the core principles of Magento's backend coding is "composition over inheritance"
This means that you should try to avoid making extensions by using php "extends" whenever possible. And rely on separate implementations that you inject into Magento classes via constructor. And if their logic needs to be modified use virtual types, plugins and other methods instead.

In this case the token validation does not strictly require the validation class to extend the existing repository. I would recommend you rename the implementation to a "Model\Validation\CustomerToken" or something similarly descriptive and completely drop this interface and the "extends" call in the implementation.

You can find more information about composition over inheritance from : https://developer.adobe.com/commerce/php/best-practices/extensions/#composition-over-inheritance 
 
 If you'd like more information about this we can schedule a meeting and talk a littlebit about Magento's coding practices.

Reply:
Okay, I will change it

Reply:
@tommi-martin I change it. I create new class Model\Validation\CustomerData and put 2 methods in it. For validation token and subscription for customer.